tokio-util/src/io/sink_writer.rs

```diff
@@ -0,0 +1,110 @@
+use futures_sink::Sink;
+
+use pin_project_lite::pin_project;
+use std::io;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tokio::io::AsyncWrite;
+
+pin_project! {
+    /// Convert a [`Sink`] of byte chunks into an [`AsyncWrite`].
+    ///
+    /// Each write to the `SinkWriter` is converted into a send of `&[u8]` to the `Sink`. Flushes and shutdown are propagated to the sink's flush and close methods.
+    ///
+    /// This adapter implements `AsyncWrite` for `Sink<&[u8]>`. If you want to implement `Sink<_>` for `AsyncWrite`, see [`codec`]; if you need to implement `AsyncWrite` for `Sink<Bytes>`, see [`CopyToBytes`].
```

Comment:
Please reflow this to fit 80 columns - that way, I can read the comment easily in my editor even if I have multiple source files open side-by-side.